Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -1231,56 +1229,58 @@ define([
         OPAQUE_AND_TRANSLUCENT : 2
     };
 
-    function updateDerivedCommands(derivedCommands, command) {
-        for (var name in derivedCommands) {
-            if (derivedCommands.hasOwnProperty(name)) {
-                var derivedCommand = derivedCommands[name];
-                derivedCommand.castShadows = command.castShadows;
-                derivedCommand.receiveShadows = command.receiveShadows;
-                derivedCommand.primitiveType = command.primitiveType;
-            }
+    Cesium3DTileBatchTable.prototype.addDerivedCommands = function(frameState, commandStart) {
+        if (!frameState.passes.render) {
+            return;
+        }
+
+        var commandList = frameState.commandList;
+        var commandLength = commandList.length;
+        if (commandStart === commandLength) {
```

Comment:
This isn't super obvious.  Does it belong here?  Would it be better to do this check outside of this function?  Or at least comment this.